plugins/pulp_docker/plugins/registry.py

```diff
@@ -503,10 +504,10 @@ def _get_path(self, path, headers=None):
                     _logger.debug(_('Download unauthorized, attempting to retrieve a token.'))
                     self.token = auth_util.request_token(self.auth_downloader, request,
                                                          auth_header, self.name)
-                    request.headers = auth_util.update_token_auth_header(request.headers,
-                                                                         self.token)
-                    report = self.downloader.download_one(request)
-
+                    if not isinstance(self.token, DownloadReport):
```

Comment:
if `request_token` returns a report that contains the token, rather than the token itself, we can check if the download failed instead of checking the type of the return value.

Reply:
as far as i know this cannot happen.
token is not contained in the report unless it's a successful one.

Reply:
What I am proposing is that this line becomes:

    if report.state == report.DOWNLOAD_FAILED: